integrated_channels/integrated_channel/models.py

```diff
@@ -123,6 +176,17 @@ def __init_subclass__(cls, **kwargs):
         super().__init_subclass__(**kwargs)
         models.signals.post_save.connect(set_default_display_name, sender=cls)
 
+    def delete(self):
```

Comment:
very weird behavior here related to how Django handles [inheritance and model managers](https://docs.djangoproject.com/en/4.1/topics/db/managers/). But I couldn't get subclassed channel configs to use the custom manager's delete override. This means that when querying `BlackboardEnterpriseCustomerConfig`, you can use `objects` and `all_objects` but when calling `delete()` on a record, it would grab Django's base `Model` class' usage of `delete` instead of the one defined in the soft delete manager. To remedy this I just overrode/added the three required soft delete methods to the base config.